<launch>
  <arg name="verbose_planner" default="false"/>
  <arg name="rviz_config_voxblox" default="example_full.rviz"/>
    
  <!-- Voxblox arguments -->
  <arg name="voxel_size" default="0.05"/>
  <arg name="voxels_per_side" default="16"/>
  <arg name="method" default="fast"/>
  <arg name="verbose_voxblox" default="false"/>
  
  
  <!-- Voxblox -->
	<node name="voxblox_node" pkg="voxblox_ros" type="esdf_server" output="screen" args="-alsologtostderr">
		<remap from="pointcloud" to="/unreal/ue_sensor_out"/>
	  <param name="publish_esdf_map" value="true" />
	  <param name="publish_pointclouds" value="true" />
		<param name="tsdf_voxel_size" value="$(arg voxel_size)" />
		<param name="tsdf_voxels_per_side" value="$(arg voxels_per_side)" />
		<param name="color_mode" value="color" />
		<param name="use_tf_transforms" value="true" />
		<param name="update_mesh_every_n_sec" value="1.0" />
		<param name="output_mesh_as_pointcloud" value="true" />
		<param name="method" value="$(arg method)" />
		<param name="max_ray_length" value="10.0" />
		<param name="allow_clear" value="true" />
		<param name="clear_sphere_for_planning" value="false" />
		<param name="verbose" value="$(arg verbose_voxblox)" />
	</node>
	
	<!-- Random walk planner -->
	<node name="active_3d_planner_node" pkg="mav_active_3d_planning" type="active_3d_planner_node" output="screen">
		<remap from="active_3d_planner_node/esdf_map_in" to="/voxblox_node/esdf_map_out" />
		<param name="use_tf_transforms" value="true" />
		<param name="tsdf_voxel_size" value="$(arg voxel_size)" />
		<param name="tsdf_voxels_per_side" value="$(arg voxels_per_side)" />
		<param name="update_mesh_every_n_sec" value="0.0" />
		
		<!--remap from="odometry" to="odometry_sensor1/odometry" /-->
		<param name="verbose" value="$(arg verbose_planner)" />
	</node>
	
	<!-- Visualization -->
	<node type="rviz" name="rviz_voxblox" pkg="rviz" args="-d $(find unreal_cv_ros)/cfg/$(arg rviz_config_voxblox)"/>
	
	<!-- Voxblox map-->
	<node pkg="rosservice" type="rosservice" name="load_map_service" args="call /voxblox_node/load_map {'file_path': '/home/lukas/Documents/MT/test_map'}"/> 
	
</launch>
